crates/red_knot_python_semantic/src/types/builder.rs

```diff
@@ -53,7 +60,12 @@ impl<'db> UnionBuilder<'db> {
                     self = self.add(*element);
                 }
             }
+            // Adding `Never` to a union is a no-op.
             Type::Never => {}
+            // Adding `object` to a union results in `object`.
+            ty if ty.is_object(self.db) => {
```

Comment:
Could also be
```suggestion
            Type::Instance(instance) if instance.class.is_object(self.db) => {
```
but I somehow doubt that it would be much faster(?)

Reply:
I wondered the same thing; I think it could be unpacked even further to match against `class.known` of the `InstanceType` and eliminate the `if` entirely? But in the absence of evidence that that is faster, it just looks more verbose.